### PR TITLE
Allow comma separated packages for ]dev

### PR DESCRIPTION
**Pull request by mhauru**
In line with how e.g. `]add` works.

### CODE REVIEW

File: test/repl.jl [hunk 1]
```diff
@@ -122,6 +122,8 @@ temp_pkg_dir(;rm=false) do project_path; cd(project_path) do;
     mktempdir() do tmp_dev_dir
     withenv("JULIA_PKG_DEVDIR" => tmp_dev_dir) do
         pkg"develop Example"
+        pkg"develop Example,PackageCompiler"
+        pkg"develop Example PackageCompiler"
```
Comment (IanButterworth) [inline test/repl.jl:126]:
```suggestion
        pkg"develop Example1, Example2"
        pkg"develop Example1 Example2"
```

Comment (mhauru) [reply test/repl.jl:126]: I thought the use of `Example` relied on this package: https://github.com/JuliaLang/Example.jl i.e. that these had to actually be real packages. Very happy to change this if that's not the case, the use of PackageCompiler here is very clunky.

Comment (IanButterworth) [reply test/repl.jl:126]: It's not a big deal. `Package1 Package2` also works

Comment (IanButterworth) [reply test/repl.jl:126]: I missed that this was in tests. Thought it was the docstring.. whoops. 

LGTM then

**Comment by IanButterworth** [comment]
I'm marking this for backport as the omission of this seems like a bug

**Pull request merged**